openapi.yaml

```diff
@@ -0,0 +1,2411 @@
+
+            This can be combined with execution_date_lte parameter to receive only the selected period.
+        execution_date_lte:
+          type: string
+          format: date-time
+          description: >
+            Returns objects less than or equal to the specified date.
+
+            This can be combined with execution_date_gte parameter to receive only the selected period.
+        start_date_gte:
+          type: string
+          format: date-time
+          description: >
+            Returns objects greater or equal the specified date.
+
+            This can be combined with startd_ate_lte parameter to receive only the selected period.
+        start_date_lte:
+          type: string
+          format: date-time
+          description: >
+            Returns objects less or equal the specified date.
+
+            This can be combined with start_date_gte parameter to receive only the selected period.
+        end_date_gte:
+          type: string
+          format: date-time
+          description: >
+            Returns objects greater or equal the specified date.
+
+            This can be combined with start_date_lte parameter to receive only the selected period.
+        end_date_lte:
+          type: string
+          format: date-time
+          description: >
+            Returns objects less than or equal to the specified date.
+
+            This can be combined with start_date_gte parameter to receive only the selected period.
+        duration_gte:
+          type: number
+          description: >
+            Returns objects greater than or equal to the specified values.
+
+            This can be combined with duration_lte parameter to receive only the selected period.
+        duration_lte:
+          type: number
+          description: >
+            Returns objects less than or equal to the specified values.
+
+            This can be combined with duration_gte parameter to receive only the selected range.
+        State:
+          type: array
+          items:
+            type: string
+          description:
+            The value can be repeated to retrieve multiple matching values (OR condition).
+        Pool:
+          type: array
+          items:
+            type: string
+          description:
+            The value can be repeated to retrieve multiple matching values (OR condition).
+        Queue:
+          type: array
+          items:
+            type: string
+          description:
+            The value can be repeated to retrieve multiple matching values (OR condition).
+
+    # Common data type
+    ScheduleInterval:
+      oneOf:
+        - $ref: '#/components/schemas/TimeDelta'
+        - $ref: '#/components/schemas/RelativeDelta'
+        - $ref: '#/components/schemas/CronExpression'
+      discriminator:
+        propertyName: objectType
+
+    TimeDelta:
+      type: object
+      properties:
+        objectType: {type: string}
```

Comment:
This seems a bit odd here in the responses to have this objectType:

```json
      "execution_timeout": {
        "objectType": "string",
        "days": 0,
        "seconds": 0,
        "microsecond": 0
      }
```

Also I could do `{ objectType: "carrot", days: 1 }` here :) Does OpenAPI allow the Enum or Const type from JSONSchema?

Reply:
Oh also, case on `objectType` -> `object_type, and `__object_type`, or `$object_type` would be a better way of separating this "type" param from the actual data. `__type` was what we used in the Serialized Dag representation, so if you used that here to we could then return it without needing any translation.

Reply:
> Does OpenAPI allow the Enum or Const type from JSONSchema?

This is described on this page. 
https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/
I do not see such a possibility. I can only add the mapping property to further emphasize the corresponding mapping rules.

Reply:
Do we not also need `required: objectType` as that example shows?

Reply:
https://swagger.io/docs/specification/data-models/keywords/ -- enum is supported.

Reply:
OpenAPI support enum as a part of the specification.  You can define enum in specification, but you can't use an enum to define the specification.  There is no special syntax in YAML that would allow defining references to these elements in other parts of the specification.

Reply:
I will change it to __type.

Reply:
I introduced changes. Commits:
https://github.com/apache/airflow/pull/8721/commits/ffbb8985e1754946bcbc15f3bc57f251a3a014ab
https://github.com/apache/airflow/pull/8721/commits/c0356efa03943058dc39d1cf88937da2ee8bf53b